resource/ui/ts/pages/nodes.ts

```diff
@@ -0,0 +1,57 @@
+// source: pages/nodes.ts
+/// <reference path="../typings/mithriljs/mithril.d.ts" />
+/// <reference path="../models/node_status.ts" />
+
+// Author: Bram Gruneir (bram.gruneir@gmail.com)
+
+/**
+ * AdminViews is the primary module for Cockroaches administrative web
+ * interface.
+ */
+module AdminViews {
+  /**
+   * Nodes is the view for exploring the status of all nodes.
+   */
+  export module Nodes {
+    /**
+     * Model holds stuff.
+     */
+    module Model {
+      export var nodeStatuses: Models.NodeStatus.Nodes = new Models.NodeStatus.Nodes();
+    }
+
+    export module Page {
+      export function controller() { }
+      export function view() {
+        Model.nodeStatuses.QueryAll()
+        return m("div", [
+          m("h2", "Nodes Status"),
+          m("div", [
+            m("h3", "Nodes"),
+            m("ul", [
+              Object.keys(Model.nodeStatuses.desc()).sort().map(function(nodeId) {
+                var desc = Model.nodeStatuses.desc()[nodeId].desc;
+                return m("li", { key: desc.node_id }, "ID:" + desc.node_id + " Address:" + desc.address.network + "-" + desc.address.address)
+              }),
+            ]),
+          ])
+        ]);
+      }
+    }
+
+    export module SinglePage {
+      export function controller() { }
+      export function view() {
+        var nodeId = m.route.param("node_id");
+        Model.nodeStatuses.QueryOne(nodeId)
```

Comment:
This is the biggest issue - you should never issue queries in a view.  Each "view" method is called after every single data change that mithril knows about - that is, any completed network request (if called through m.request) or after any event handler hooked up via an m() object.

View methods need to be cheap, and should basically only be generating a set of m() objects.  Querying should be managed in the controller.